### PR TITLE
add more alt-svc test coverage

### PR DESCRIPTION
**Pull request by xquery**
add some tests that exercise alt-svc lookup for http->http/2 and https>http/2 scenarios.

### CODE REVIEW

File: tests/data/test358 [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+# Server-side
+<reply>
+<data nocheck="yes">
+HTTP/1.1 200 OK
+Date: Thu, 09 Nov 2010 14:49:00 GMT
+Content-Length: 6
+Connection: close
+Content-Type: text/html
+Funny-head: yesyes
+Alt-Svc: h2=":9014", ma=315360000; persist=0
```
Comment (bagder) [inline tests/data/test358:21]: Port 9014 is not guaranteed to run the HTTP/2 server. Use the `%HTTP2PORT` variable here instead! (works since d009bc2e56 landed) Should apply for all appearances of 9014 in these test cases. You can probably verify that it works by using `runtests.pl -b 9000` to move the base port.

This is also the reason why the alt-svc CI tests failed on travis just now.

File: tests/data/test358 [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+</server>
+ <name>
+HTTPS GET translated by alt-svc lookup to HTTP/2 GET
+ </name>
+<setenv>
+# make debug-curl accept Alt-Svc over plain HTTP
+CURL_ALTSVC_HTTP="yeah"
+</setenv>
+ <command>
+--http2 --alt-svc "log/altsvc-358" "http://%HOSTIP:%HTTPPORT/358" "http://%HOSTIP:%HTTPPORT/358"
```
Comment (bagder) [inline tests/data/test358:46]: Should we really need to have `--http2` specified to get this to work? I understand you need it now, but isn't this actually a bug you hide by using this flag? Goes for both test cases.

**Comment by xquery** [comment]
hmmm, updating to latest master and I now get:

'Use of uninitialized value $HTTPPORT in substitution iterator at ./runtests.pl line 3210.'

I think thats related to the recent test changes ... let me see how I have to update these tests.

File: tests/data/test358 (hunk 2 above)
Comment (xquery) [reply tests/data/test358:46]: yes, I was thinking (pedantically) that my bug fix would come with new PR fixing code and these tests ... happy to bundle all of em together.

File: tests/data/test358 (hunk 1 above)
Comment (xquery) [reply tests/data/test358:21]: yes, though I initially found var substitution did not work in data section .. let me try again.

File: tests/data/test358 (hunk 2 above)
Comment (bagder) [reply tests/data/test358:46]: It just felt weird to land tests that verifies the wrong behavior, but you're right we can also view it as verifying *current* behavior and then change that in a subsequent bug-fix. I'll be okay with that if you prefer that way.

**Comment by bagder** [comment]
> Use of uninitialized value

That's a bug of course, but happens because `$HTTPPORT` is not initialized for you and it really should be at that point since the http server should've been started. I would expect!

**Comment by bagder** [comment]
0f5db7b263 should fix the warning

**Comment by xquery** [comment]
FYI - some tests failed eg. 1700 1701 1702 failing which is unrelated.

**Comment by xquery** [comment]
These tests characterise current behaviour ... which is somewhat broken (we should not have to use --http2). The fix for this will be included in https://github.com/curl/curl/issues/4908

**Comment by bagder** [comment]
@xquery nits on the merge:

1. squash fixes before merge to the minimum amount of logical commit
2. follow the commit message template, most importantly `[area]: [short desc]` on the first line and a "Closes #[PR]" in the body

**Comment by xquery** [comment]
apologies, lazy on my part - FWIW the squash option is disabled in curl/curl github though I will make sure to squash commits in the branch first going forward.

**Comment by bagder** [comment]
Yes, we strongly discourage the use of the UI buttons for merge anyway as they very rarely do what we want.
